### PR TITLE
Added descriptive comments to DSPASpec

### PR DESCRIPTION
**Pull request by hbelmiro**
## The issue resolved by this Pull Request:
Resolves https://github.com/opendatahub-io/data-science-pipelines-operator/issues/200

## Description of your changes:
Added descriptive comments to DSPASpec

## Testing instructions
1. Deploly DSPO/DSPA
2. Navigate to DSPA in OCP console API Explorer, mouse over the affected fields and confirm the descriptions shows up.

## Checklist
- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by dsp-developers** [comment]
A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-360`
An OCP cluster where you are logged in as cluster admin is required.

To use this image run the following:

```bash
cd $(mktemp -d)
git clone git@github.com:opendatahub-io/data-science-pipelines-operator.git
cd data-science-pipelines-operator/
git fetch origin pull/360/head
git checkout -b pullrequest 1104d1815ea10e70695ad29da591ae36fe051b16
make deploy IMG="quay.io/opendatahub/data-science-pipelines-operator:pr-360"
```

More instructions [here](https://github.com/opendatahub-io/data-science-pipelines-operator#deploy-dsp-instance) on how to deploy and test a Data Science Pipelines Application.

File: api/v1alpha1/dspipeline_types.go [hunk 1]
```diff
@@ -22,62 +22,78 @@ import (
...
 	// +kubebuilder:validation:Optional
 	ArchiveLogs              bool   `json:"archiveLogs"`
 	ArtifactImage            string `json:"artifactImage,omitempty"`
 	CacheImage               string `json:"cacheImage,omitempty"`
 	MoveResultsImage         string `json:"moveResultsImage,omitempty"`
 	*ArtifactScriptConfigMap `json:"artifactScriptConfigMap,omitempty"`
+	// Default: true
 	// +kubebuilder:default:=true
 	// +kubebuilder:validation:Optional
 	InjectDefaultScript bool `json:"injectDefaultScript"`
```
Comment (HumairAK) [inline api/v1alpha1/dspipeline_types.go:68]: "Inject the archive step script."

File: api/v1alpha1/dspipeline_types.go [hunk 2]
```diff
@@ -22,62 +22,78 @@ import (
...
 	// +kubebuilder:default:=120
 	DBConfigConMaxLifetimeSec int `json:"dbConfigConMaxLifetimeSec,omitempty"`
+	// Default: true
 	// +kubebuilder:default:=true
 	// +kubebuilder:validation:Optional
 	CollectMetrics bool `json:"collectMetrics"`
+	// Default: true
 	// +kubebuilder:default:=true
 	// +kubebuilder:validation:Optional
 	EnableRoute bool `json:"enableOauth"`
```
Comment (HumairAK) [inline api/v1alpha1/dspipeline_types.go:91]: "Create an Openshift Route for this DSP API Server."

File: api/v1alpha1/dspipeline_types.go [hunk 3]
```diff
@@ -22,62 +22,78 @@ import (
...
+	// Default: true
 	// +kubebuilder:default:=true
 	// +kubebuilder:validation:Optional
 	EnableSamplePipeline bool `json:"enableSamplePipeline"`
```
Comment (HumairAK) [inline api/v1alpha1/dspipeline_types.go:95]: "Include sample pipelines with the deployment of this DSP API Server."

File: api/v1alpha1/dspipeline_types.go [hunk 4]
```diff
@@ -90,26 +106,31 @@ type ArtifactScriptConfigMap struct {
 }
 
 type PersistenceAgent struct {
+	// Default: true
 	// +kubebuilder:default:=true
 	// +kubebuilder:validation:Optional
 	Deploy bool   `json:"deploy"`
 	Image  string `json:"image,omitempty"`
```
Comment (HumairAK) [inline api/v1alpha1/dspipeline_types.go:113]: "Specify a custom image for DSP PersistenceAgent."

File: api/v1alpha1/dspipeline_types.go [hunk 5]
```diff
@@ -90,26 +106,31 @@ type ArtifactScriptConfigMap struct {
 }
 
 type PersistenceAgent struct {
+	// Default: true
 	// +kubebuilder:default:=true
 	// +kubebuilder:validation:Optional
 	Deploy bool   `json:"deploy"`
```
Comment (HumairAK) [inline api/v1alpha1/dspipeline_types.go:112]: "Enable DS Pipelines Operator management of Persisence Agent. Setting Deploy to false disables operator reconciliation."

File: api/v1alpha1/dspipeline_types.go [hunk 6]
```diff
@@ -90,26 +106,31 @@ type ArtifactScriptConfigMap struct {
...
 	Image  string `json:"image,omitempty"`
+	// Default: 2
 	// +kubebuilder:default:=2
 	NumWorkers int                   `json:"numWorkers,omitempty"`
```
Comment (HumairAK) [inline api/v1alpha1/dspipeline_types.go:116]: "Number of worker for Persistence Agent sync job."

File: api/v1alpha1/dspipeline_types.go [hunk 7]
```diff
@@ -90,26 +106,31 @@ type ArtifactScriptConfigMap struct {
...
 	Resources  *ResourceRequirements `json:"resources,omitempty"`
```
Comment (HumairAK) [inline api/v1alpha1/dspipeline_types.go:117]: "Specify custom Pod resource requirements for this component."

File: api/v1alpha1/dspipeline_types.go [hunk 8]
```diff
@@ -90,26 +106,31 @@ type ArtifactScriptConfigMap struct {
...
 type ScheduledWorkflow struct {
+	// Default: true
 	// +kubebuilder:default:=true
 	// +kubebuilder:validation:Optional
 	Deploy bool   `json:"deploy"`
 	Image  string `json:"image,omitempty"`
+	// Default: UTC
 	// +kubebuilder:default:=UTC
 	CronScheduleTimezone string                `json:"cronScheduleTimezone,omitempty"`
 	Resources            *ResourceRequirements `json:"resources,omitempty"`
```
Comment (HumairAK) [inline api/v1alpha1/dspipeline_types.go:129]: "Specify custom Pod resource requirements for this component."

File: api/v1alpha1/dspipeline_types.go [hunk 9]
```diff
@@ -122,25 +143,28 @@ type MlPipelineUI struct {
...
 	PasswordSecret *SecretKeyValue `json:"passwordSecret,omitempty"`
 	// +kubebuilder:default:=mlpipeline
-	// The database name that will be created. Should match `^[a-zA-Z0-9_]+`
+	// The database name that will be created. Should match `^[a-zA-Z0-9_]+`. // Default: mlpipeline
 	// +kubebuilder:validation:Pattern=`^[a-zA-Z0-9_]+$`
 	DBName string `json:"pipelineDBName,omitempty"`
+	// Default: 10Gi
 	// +kubebuilder:default:="10Gi"
 	PVCSize   resource.Quantity     `json:"pvcSize,omitempty"`
 	Resources *ResourceRequirements `json:"resources,omitempty"`
```
Comment (HumairAK) [inline api/v1alpha1/dspipeline_types.go:170]: "Specify custom Pod resource requirements for this component."

File: api/v1alpha1/dspipeline_types.go [hunk 10]
```diff
@@ -158,18 +182,22 @@ type ExternalDB struct {
...
 	// +kubebuilder:validation:Optional
 	Deploy bool `json:"deploy"`
+	// Default: mlpipeline
 	// +kubebuilder:default:=mlpipeline
 	Bucket              string `json:"bucket,omitempty"`
 	*S3CredentialSecret `json:"s3CredentialsSecret,omitempty"`
+	// Default: 10Gi
 	// +kubebuilder:default:="10Gi"
 	PVCSize   resource.Quantity     `json:"pvcSize,omitempty"`
 	Resources *ResourceRequirements `json:"resources,omitempty"`
```
Comment (HumairAK) [inline api/v1alpha1/dspipeline_types.go:203]: "Specify custom Pod resource requirements for this component."

File: api/v1alpha1/dspipeline_types.go [hunk 11]
```diff
@@ -90,26 +106,31 @@ type ArtifactScriptConfigMap struct {
...
 	// +kubebuilder:default:=2
 	NumWorkers int                   `json:"numWorkers,omitempty"`
 	Resources  *ResourceRequirements `json:"resources,omitempty"`
 }
 
 type ScheduledWorkflow struct {
+	// Default: true
 	// +kubebuilder:default:=true
 	// +kubebuilder:validation:Optional
 	Deploy bool   `json:"deploy"`
```
Comment (HumairAK) [inline api/v1alpha1/dspipeline_types.go:124]: "Enable DS Pipelines Operator management of ScheduledWorkflow. Setting Deploy to false disables operator reconciliation."

File: api/v1alpha1/dspipeline_types.go [hunk 12]
```diff
@@ -90,26 +106,31 @@ type ArtifactScriptConfigMap struct {
...
 	Image  string `json:"image,omitempty"`
```
Comment (HumairAK) [inline api/v1alpha1/dspipeline_types.go:125]: "Specify a custom image for DSP ScheduledWorkflow controller."

File: api/v1alpha1/dspipeline_types.go [hunk 13]
```diff
@@ -90,26 +106,31 @@ type ArtifactScriptConfigMap struct {
...
+	// Default: UTC
 	// +kubebuilder:default:=UTC
 	CronScheduleTimezone string                `json:"cronScheduleTimezone,omitempty"`
```
Comment (HumairAK) [inline api/v1alpha1/dspipeline_types.go:128]: "Specify the Cron timezone used for ScheduledWorkflow PipelineRuns."

File: api/v1alpha1/dspipeline_types.go [hunk 14]
```diff
@@ -90,26 +106,31 @@ type ArtifactScriptConfigMap struct {
...
 	Resources            *ResourceRequirements `json:"resources,omitempty"`
 }
 
 type MlPipelineUI struct {
+	// Default: true
 	// +kubebuilder:default:=true
 	// +kubebuilder:validation:Optional
 	Deploy        bool                  `json:"deploy"`
```
Comment (HumairAK) [inline api/v1alpha1/dspipeline_types.go:136]: "Enable DS Pipelines Operator management of KFP UI. Setting Deploy to false disables operator reconciliation."

File: api/v1alpha1/dspipeline_types.go [hunk 15]
```diff
@@ -122,25 +143,28 @@ type MlPipelineUI struct {
...
 	// +kubebuilder:default:=false
 	// +kubebuilder:validation:Optional
 	DisableHealthCheck bool `json:"disableHealthCheck"`
 }
 
 type MariaDB struct {
+	// Default: true
 	// +kubebuilder:default:=true
 	// +kubebuilder:validation:Optional
 	Deploy bool   `json:"deploy"`
```
Comment (HumairAK) [inline api/v1alpha1/dspipeline_types.go:156]: "Enable DS Pipelines Operator management of MariaDB. Setting Deploy to false disables operator reconciliation."

File: api/v1alpha1/dspipeline_types.go [hunk 16]
```diff
@@ -158,18 +182,22 @@ type ExternalDB struct {
...
 	// +kubebuilder:default:=false
 	// +kubebuilder:validation:Optional
 	DisableHealthCheck bool `json:"disableHealthCheck"`
 }
 
 type Minio struct {
+	// Default: true
 	// +kubebuilder:default:=true
 	// +kubebuilder:validation:Optional
 	Deploy bool `json:"deploy"`
```
Comment (HumairAK) [inline api/v1alpha1/dspipeline_types.go:195]: "Enable DS Pipelines Operator management of Minio. Setting Deploy to false disables operator reconciliation."

File: api/v1alpha1/dspipeline_types.go [hunk 17]
```diff
@@ -178,6 +206,7 @@ type Minio struct {
 }
 
 type MLMD struct {
+	// Default: false
 	// +kubebuilder:default:=false
 	// +kubebuilder:validation:Optional
 	Deploy  bool `json:"deploy"`
```
Comment (HumairAK) [inline api/v1alpha1/dspipeline_types.go:212]: "Enable DS Pipelines Operator management of MLMD. Setting Deploy to false disables operator reconciliation."

File: api/v1alpha1/dspipeline_types.go [hunk 18]
```diff
@@ -122,25 +143,28 @@ type MlPipelineUI struct {
...
 	// +kubebuilder:validation:Optional
 	DisableHealthCheck bool `json:"disableHealthCheck"`
 }
 
 type MariaDB struct {
+	// Default: true
 	// +kubebuilder:default:=true
 	// +kubebuilder:validation:Optional
 	Deploy bool   `json:"deploy"`
 	Image  string `json:"image,omitempty"`
```
Comment (HumairAK) [inline api/v1alpha1/dspipeline_types.go:157]: "Specify a custom image for DSP MariaDB pod."

Comment (HumairAK) [reply api/v1alpha1/dspipeline_types.go:157]: can you also add similar comments for the remaining "image" fields

File: api/v1alpha1/dspipeline_types.go [hunk 19]
```diff
@@ -122,25 +143,28 @@ type MlPipelineUI struct {
...
 	Username       string          `json:"username,omitempty"`
 	PasswordSecret *SecretKeyValue `json:"passwordSecret,omitempty"`
 	// +kubebuilder:default:=mlpipeline
-	// The database name that will be created. Should match `^[a-zA-Z0-9_]+`
+	// The database name that will be created. Should match `^[a-zA-Z0-9_]+`. // Default: mlpipeline
 	// +kubebuilder:validation:Pattern=`^[a-zA-Z0-9_]+$`
 	DBName string `json:"pipelineDBName,omitempty"`
+	// Default: 10Gi
 	// +kubebuilder:default:="10Gi"
 	PVCSize   resource.Quantity     `json:"pvcSize,omitempty"`
```
Comment (HumairAK) [inline api/v1alpha1/dspipeline_types.go:169]: Customize the size of the PVC created for the default MariaDB instance.

File: api/v1alpha1/dspipeline_types.go [hunk 20]
```diff
@@ -158,18 +182,22 @@ type ExternalDB struct {
...
 	// +kubebuilder:default:=true
 	// +kubebuilder:validation:Optional
 	Deploy bool `json:"deploy"`
+	// Default: mlpipeline
 	// +kubebuilder:default:=mlpipeline
 	Bucket              string `json:"bucket,omitempty"`
 	*S3CredentialSecret `json:"s3CredentialsSecret,omitempty"`
+	// Default: 10Gi
 	// +kubebuilder:default:="10Gi"
 	PVCSize   resource.Quantity     `json:"pvcSize,omitempty"`
```
Comment (HumairAK) [inline api/v1alpha1/dspipeline_types.go:202]: Customize the size of the PVC created for the Minio instance.

File: api/v1alpha1/dspipeline_types.go [hunk 21]
```diff
@@ -158,18 +182,22 @@ type ExternalDB struct {
...
 }
 
 type Minio struct {
+	// Default: true
 	// +kubebuilder:default:=true
 	// +kubebuilder:validation:Optional
 	Deploy bool `json:"deploy"`
+	// Default: mlpipeline
 	// +kubebuilder:default:=mlpipeline
 	Bucket              string `json:"bucket,omitempty"`
```
Comment (HumairAK) [inline api/v1alpha1/dspipeline_types.go:198]: Provide the Bucket name that will be used to store artifacts in S3. If provided bucket does not exist, DSP Apiserver will attempt to create it. As such the credentials provided should have sufficient permissions to do create buckets.

File: api/v1alpha1/dspipeline_types.go [hunk 22]
```diff
@@ -158,18 +182,22 @@ type ExternalDB struct {
...
 	*S3CredentialSecret `json:"s3CredentialsSecret,omitempty"`
```
Comment (HumairAK) [inline api/v1alpha1/dspipeline_types.go:199]: Credentials for the S3 user (e.g. IAM user cred stored in a k8s secret.).
Note that the S3 user should have the permissions to create a bucket if the provided bucket does not exist.

File: api/v1alpha1/dspipeline_types.go (hunk 10 above)
Comment (HumairAK) [reply api/v1alpha1/dspipeline_types.go:203]: can you add similar comments to all other `Resources` fields.

**Review by HumairAK — CHANGES_REQUESTED** [review]

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-360`

**Comment by hbelmiro** [comment]
@HumairAK I incorporated your suggestions. Can you have a second look, please?

**Comment by HumairAK** [comment]
@hbelmiro the `make manifests` command should not result in any diffs. Can you double check?

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-360`

**Comment by HumairAK** [comment]
/lgtm
/approve

Great work!

**Pull request merged**